### PR TITLE
feat(api): add v2.1 api documentation.

### PR DESCRIPTION
**Pull request by Joshua-Anderson**
@helgi Did I overlook any API changes for this release (see the `What's New` section)
I just skimmed at the [commit diff between v2.0.0 and master](https://github.com/deis/controller/compare/v2.0.0...master?expand=1).

### CODE REVIEW

**Comment by helgi** [comment]
`healthcheck` is a new field within the `Config` endpoint, i see you added that in the examples but not what's new but I suppose we never really added any granularity if we have nested structures

**Comment by Joshua-Anderson** [comment]
I added a message about the `healthchecks` field and removed `X-Deis-Release` headers that were entirely removed from workflow.

File: src/reference-guide/controller-api/v2.1.md [hunk 1]
```diff
@@ -0,0 +1,1409 @@
+# Controller API v2.1
+
+This is the v2.1 REST API for the Controller.
+
+## What's New
+
+**New!** `healthchecks` field in configuration, deprecates the `HEALTHCHECK_*` environmental variables.
```
Comment (helgi) [inline src/reference-guide/controller-api/v2.1.md]: It needs to be singular

Comment (bacongobbler) [inline src/reference-guide/controller-api/v2.1.md]: s/environmental/environment :)

**Comment by Joshua-Anderson** [comment]
Fixed comments on healthchecks.

**Comment by krancour** [comment]
@Joshua-Anderson shouldn't there also be a change in `mkdocs.yml` to add this to the doc tree?

**Comment by bacongobbler** [comment]
I'm surprised that mkdocs didn't catch that, but yes you are correct @krancour.

**Comment by Joshua-Anderson** [comment]
@krancour Fixed!

**Pull request merged**